enarx-keep-sgx-shim/src/start.S

```diff
@@ -154,6 +188,27 @@ enclave:
     .globl syscall
     .type syscall, @function
 syscall:
+    # Save stack pointer, allocate space, and set
+    # 64-byte alignment on stack for xsavec
+    mov     %rsp,                   %rbx
+    sub     $4096,                  %rsp
+    and     $(-0x40),               %rsp
+
+    # Zero the xsave header area on stack
+    movq    $0,        512(%rsp)
+    movq    $0,        520(%rsp)
+    movq    $0,        528(%rsp)
+    movq    $0,        536(%rsp)
+    movq    $0,        544(%rsp)
+    movq    $0,        552(%rsp)
+    movq    $0,        560(%rsp)
+    movq    $0,        568(%rsp)
+
+    # Set mask and perform xsavec
+    movq    $~0,                   %rdx
+    movq    $~0,                   %rax
+    xsavec (%rsp)
+
     savep                                           # Save preserved registers
```

Comment:
With `savep` you saved %rbx, but the original contents was destroyed earlier by `mov     %rsp,                   %rbx`.

So, I would suggest `savep` after `syscall:` immediately and `pushq %rbx` here instead.

You really should check, that all is restored in reverse order...

syscall:
1. savep
2. rsp -> rbx
3. xsave
4. push rbx aka rsp from 2.

syscall return:
4. pop rbx aka rsp for 2.
3. xrstor
2. rbx -> rsp
1. loadp